model/channel_sidebar.go

```diff
@@ -109,3 +110,13 @@ func (o OrderedSidebarCategories) ToJson() []byte {
 		return b
 	}
 }
+
+func IsValidCategoryId(s string) bool {
+	// Category IDs can either be regular IDs
+	if IsValidId(s) {
+		return true
+	}
+
+	// Or default categories can follow the pattern {type}_{userID}_{teamID}
+	return regexp.MustCompile("(favorites|channels|direct_messages)_[a-z0-9]{26}_[a-z0-9]{26}").MatchString(s)
```

Comment:
We should avoid creating a new instance of `regexp.MustCompile` every time. They should be created once. Creating a new instance every time has a performance cost which should be avoided.